### PR TITLE
Default to legacy format if JSON template is loaded

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

If JSON template is configured for loading, Filebeat should still default to legacy index template formats to avoid breaking existing template configurations. It also adds a warning to let users know that they have to migrate from legacy format.

## Why is it important?

Minimizes the breaking changes of moving from legacy templates to composable templates.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by kvch** [comment]
@jlind23 I have removed backport labels because this PR is opened against 7.16 and it won't be forward ported.

File: libbeat/tests/system/config/mockbeat.yml.j2 [hunk 1]
```diff
@@ -77,6 +77,7 @@ setup.template:
   name: {{ es_template_name }}
   pattern: {{ es_template_pattern }}
   overwrite: {{ template_overwrite|default("false") }}
+  type: {{ template_type|default("index") }}
```
Comment (ruflin) [inline libbeat/tests/system/config/mockbeat.yml.j2:80]: Wondering if we should add a test for a legacy template to make sure it keeps working?

File: libbeat/template/config.go [hunk 2]
```diff
@@ -76,6 +77,25 @@ func DefaultConfig() TemplateConfig {
...
+func Unpack(c *common.Config) (TemplateConfig, error) {
+	if c == nil {
+		return DefaultConfig(), nil
+	}
+
+	jsonEnabled, _ := c.Bool("json.enabled", -1)
+	config := DefaultConfig()
+	if jsonEnabled {
+		cfgwarn.Deprecate("8.0.0", "Please migrate your JSON templates from legacy template format to composable index template.")
+		config.Type = IndexTemplateLegacy
```
Comment (ruflin) [inline libbeat/template/config.go:89]: What if the user wants to use the new templates in 7.16? Will line 93 overwrite it?

**Review by ruflin — COMMENTED** [review]
Change LGTM.

Do we need a changelog entry too or similar to mark it as deprecated.

One part I stumbled over is the "JSON template" naming. Should we call it instead legacy templates or similar?

File: libbeat/template/config.go (hunk 2 above)
Comment (kvch) [reply libbeat/template/config.go:89]: No, it is not overwritten. The configuration from the user is unpacked later.

**Comment by kvch** [comment]
> Do we need a changelog entry too or similar to mark it as deprecated.

It has been deprecated previously. I have added more info to the documentation regarding how we fall back to legacy if the template is provided in a JSON file.

> One part I stumbled over is the "JSON template" naming. Should we call it instead legacy templates or similar?

What do you mean? JSON template is the template users provide in a JSON file. It can be in any format legacy or composable depending on what the user wants.

File: libbeat/template/config.go (hunk 2 above)
Comment (ruflin) [reply libbeat/template/config.go:89]: That is what I mean by 93, this is where the user config is unpacked. Looks like we are on the same page.

**Comment by ruflin** [comment]
> "Please migrate your JSON templates from legacy template format to composable index template."

This is nothing you added specificially in this PR. What we refer to as `JSON templates`  are Elasticsearch index templates in json format. I know what we refer to with `JSON templates` but I would argue for a users it could be confusing. Lets keep it as is as we haven't seen any complaints around it.

**Review by ruflin — APPROVED** [review]

**Pull request merged**